### PR TITLE
Fix/fix and cleanup async functionality

### PR DESCRIPTION
**Pull request by w4bremer**

### CODE REVIEW

**Review by zrrbite — COMMENTED** [review]
- Here are some concrete changes that could be made to README.md
- The $ModuleName change to the function in anonymous namespace seems like the right thing to do in light of Unreals "unity" build style - but i'm not sure i can make useful comments on the rest.

File: README.md [hunk 1]
```diff
@@ -1,9 +1,42 @@
 # Epic Games Unreal Engine template
 
-An unreal engine template code generator.
+This template generates an *Unreal Engine* C++ plugin. Its main purpose is to ease the data integration into any *Unreal Engine* project.
```
[the remark this answers is not in the document]
Comment (jryannel) [reply README.md:3]: # Epic Games Unreal Engine template for ApiGear

This is a code generation template for the ApiGear integration suite. It generates an Unreal Engine C++ plugin with full blueprint support based on an API model.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
 
-## Features
+For more details on *ApiGear* please visit [apigear.io](https://apigear.io) or the [documentation](https://docs.apigear/).
 
-* api: create interface files and simulation support
-* scaffold: create a fully featured project with reference implementations and tests
-* olink: create the adaption layer for the OLink protocol
+## Plugin
+
+The generated code comes in the form of an UE plugin and thus can easily be used in any UE project.
```
[the remark this answers is not in the document]
Comment (jryannel) [reply README.md:10]: I would add after each feature the name "feature" so "Plugin feature" it makes things more clear.

And then start the sentence with: The "plugin" feature ...

File: README.md [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+
+The generated plugin code is regularly tested in these configurations:
+| platform | UE4.25 | UE4.27 | UE5.1 |
+|----------| :---: | :---: | :---: |
+| Windows  |:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+| Linux    |:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+## Code Generation Features
+
+The template offers the following feature switches which can be enabled during the code generation:
+* api: create interface files and an abstract implementation
```
Comment (jryannel) [inline README.md:38]: I would put the name of a feature in code markers (`api`) it makes it clear this is an identifier.

**Review by jryannel — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+### Compatibility
+
+The generated plugin code is regularly tested in these configurations:
+| platform | UE4.25 | UE4.27 | UE5.1 |
+|----------| :---: | :---: | :---: |
+| Windows  |:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+| Linux    |:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+## Code Generation Features
+
+The template offers the following feature switches which can be enabled during the code generation:
```
[the remark this answers is not in the document]
Comment (w4bremer) [reply README.md:37]: The documentation lacks this section, Will be added next

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (w4bremer) [reply README.md:10]: This part is about the "output" of the template.
That's why I explicitly mention "plugin" again here.
It also is section title and contains "Technologies" and "Compability"

**Pull request merged**